packages/components/src/Typography/Typography.tsx

```diff
@@ -76,23 +89,48 @@ export function Typography({
     emphasisType && emphasis[emphasisType],
     fontFamily && fontFamilies[fontFamily],
     shouldTruncateText && truncate.textTruncate,
+    underline && underlineStyles.basicUnderline,
     {
       ...(align && { [alignment[align]]: align !== `start` }),
     },
   );
 
-  let truncateLines: CSSProperties | undefined;
+  let stylesOverrides: CSSProperties = {};
 
   if (shouldTruncateText) {
-    truncateLines = {
+    stylesOverrides = {
       WebkitLineClamp: numberOfLines,
       WebkitBoxOrient: "vertical",
     };
   }
 
+  if (underline) {
+    const [underlineStyle, underlineColor] = underline.split(" ");
+
+    stylesOverrides.textDecorationStyle = underlineStyle as UnderlineStyle;
+    stylesOverrides.textDecorationColor = computeUnderlineColor(
+      underlineColor,
+      textColor,
+    );
+  }
+
   return (
-    <Tag id={id} className={className} style={truncateLines}>
+    <Tag id={id} className={className} style={stylesOverrides}>
       {children}
     </Tag>
   );
 }
+
+function computeUnderlineColor(
+  textDecorationColor: string,
+  textColor?: keyof typeof textColors,
+): string | undefined {
+  // Use underline color otherwise if a custom text color is specified, use that
```

Comment:
Nit, this comment feels a bit awkwardly worded to me. Totally non-blocking.

Reply:
@NickBoers-Jobber  Updated in 280855c889630f3a9a5a7fb74728162ca4050f70 thanks for pointing this out